samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -315,4 +315,48 @@ public void onBarrierError(String version, Throwable t) {
       stop();
     }
   }
+
+  /// listener to handle session expiration
+  public class ZkSessionStateChangedListener implements IZkStateListener {
```

Comment:
We can make this package private(No dependencies outside). 

Curious question: You could have written this in same package/anywhere. Is the main motivation for having it as inner class is to perform these actions (`coordinator.stop()`, `zkController.register|unregister`, zkutils operations) easily(not worry about constructing dependency tree for this)?